include/camp/resource.hpp

```diff
@@ -194,6 +194,7 @@ namespace resources
       EventProxy &operator=(EventProxy const &) = delete;
 
       EventProxy(Res r) : resource_{move(r)} {}
+      EventProxy(Res *r) : resource_{move(*r)} {}
```

Comment:
@trws This fixes the error I'm getting, but I'm wondering if this is ok? Should I be checking to see if the Res is movable?

Reply:
I would think neither of those should have move on them honestly.  That said, we've had some conversations about making events move only, but it shouldn't apply to the resources.  The issue is we need to deal with making sure that resources and events are correctly cleaned up, so I've been trying to root out uses of them by address or reference and remove them.  Are there a lot of uses of resources by pointer in lulesh?

Reply:
Scratch this - turns out I was accidentally using RAJA v0.13.0 which was calling the `EventProxy` constructor with a reference to the Cuda resource (which has been fixed in v0.14.0). There are no uses of resources by pointer in Lulesh, it's just calling the standard RAJA::forall which creates a Cuda resource internally by default.

Instead, there is another Camp runtime error when using RAJA v0.14.0 with the appropriate Camp v0.2.2:
[chen59@rzansel17:build]$ ./bin/lulesh-v2.0-RAJA-cuda.exe                                                                                                                                                                                                                                                    [30/1195]
Running problem size 45^3 per domain until completion
Num processors: 1
Total number of elements: 91125

To run other sizes, use -s <integer>.
To run a fixed number of iterations, use -i <integer>.
To run a more or less balanced region set, use -b <integer>.
To change the relative costs of regions, use -c <integer>.
To print out progress, use -p
To write an output file for VisIt, use -v
See help (-h) for more options

terminate called after throwing an instance of 'std::runtime_error'
  what():  campCudaErrchk(cudaStreamSynchronize(stream)) an illegal memory access was encountered /usr/workspace/wsrzc/chen59/allraja/rajaproxy/rajaproxy_git/tpl/RAJA/tpl/camp/include/camp/resource/cuda.hpp:170
Aborted

Reply:
Do you still want to have this constructor or should we close this one out?

Reply:
Whoops, meant to close this. If need be, I'll make another issue for the cudaStreamSynchronize error.